src/jgikbase/test/idmapping/storage/mongo/test_id_mapping_mongo_storage.py

```diff
@@ -323,3 +323,82 @@ def fail_get_namespace(idstorage, namespace_id, expected):
         fail('expected exception')
     except Exception as got:
         assert_exception_correct(got, expected)
+
+
+def test_add_and_remove_namespace_users(idstorage):
+    nsid = NamespaceID('foo')
+    idstorage.create_namespace(nsid)
+    assert idstorage.get_namespace(NamespaceID('foo')) == Namespace(NamespaceID('foo'), False)
+
+    idstorage.add_user_to_namespace(nsid, User(AuthsourceID('asone'), 'u1'))
+    users = set([User(AuthsourceID('asone'), 'u1')])
+    assert idstorage.get_namespace(nsid) == Namespace(NamespaceID('foo'), False, users)
+
+    idstorage.add_user_to_namespace(nsid, User(AuthsourceID('astwo'), 'u2'))
+    users.add(User(AuthsourceID('astwo'), 'u2'))
+    assert idstorage.get_namespace(nsid) == Namespace(NamespaceID('foo'), False, users)
+
+    idstorage.remove_user_from_namespace(NamespaceID('foo'), User(AuthsourceID('asone'), 'u1'))
+    users = set([User(AuthsourceID('astwo'), 'u2')])
+    assert idstorage.get_namespace(nsid) == Namespace(NamespaceID('foo'), False, users)
+
+    idstorage.remove_user_from_namespace(NamespaceID('foo'), User(AuthsourceID('astwo'), 'u2'))
+    assert idstorage.get_namespace(nsid) == Namespace(NamespaceID('foo'), False)
+
+
+def test_add_user_to_namespace_fail_inputs_None(idstorage):
+    u = User(LOCAL, 'u')
+    n = NamespaceID('n')
+    fail_add_namespace_user(idstorage, None, u, TypeError('namespace_id cannot be None'))
```

Comment:
For all these exceptions you could just use pytest.raises.

Reply:
OK. Since this is just syntactic sugar in tests I'll do it in a future PR.